### DISCUSSION TITLE
Using the right version of Ruby/JS/Etc inside a devshell

### DISCUSSION THREAD

**Discussion by dlealmt** [nix-community/nixvim#1441] [Q&A] [answered]
Hi all,

I have a DevShell where I'm using Ruby 3.3, but I noticed that the Nixvim config is using Ruby 3.1. This seems to cause Solargraph (Ruby LSP) to crash. What is the blessed way to make Nixvim use the right Ruby version when inside the DevShell? I tried using `makeNixvimWithModule` but I'm not sure if that is the right way to do it, and if it is, how do I make it find my configuration, which is outside this flake where I'm using Ruby 3.3?

Thanks,

David

**Comment by GaetanLepage** [comment, marked as answer]
Hi David,
By default, nixvim will automatically install the language server (solargraph here). If you want it to use the one from your devshell, you would need to set `plugins.lsp.servers.solagraph.package` to `null`. This would prevent nixvim from installing it.
You would then be responsible to have it available on your `$PATH`.

**Reply by dlealmt** [reply]
That was it, thanks.
